src/input/mpegts/iptv/iptv_http.c

```diff
@@ -395,14 +450,23 @@ iptv_http_complete
       if (s == NULL) {
         tvherror(LS_IPTV, "no URI in KEY attribute");
         goto end;
+      } 
+      if (memcmp(s, "../", strlen("../")) == 0) {
+        tvhtrace(LS_IPTV, "KEY URI: relative URL detected (%s), rewriting it", s);
+        absolute_key_url = merge_absolute_and_relative_urls(url, s);
+        s = absolute_key_url;
+        tvhtrace(LS_IPTV, "new KEY URI: '%s'", s);
       }
       hp->hls_encrypted = 1;
       if (strcmp(hp->hls_key_url ?: "", s)) {
         free(hp->hls_key_url);
         hp->hls_key_url = strdup(s);
         free(hp->hls_url_after_key);
         hp->hls_url_after_key = url;
-          url = strdup(s);
+        url = strdup(s);
+        if (absolute_key_url != NULL) {
+          free(absolute_key_url);
```

Comment:
It is safe to call free on NULL, so this check can be removed (unless you plan to run this code on old PalmOS ;) )